stetho/src/main/java/com/facebook/stetho/common/android/FragmentApi.java

```diff
@@ -0,0 +1,459 @@
+      return sFragmentManagerAccessor;
+    }
+
+    LogUtil.w(
+        "getFragmentManagerAccessorFor() called for unrecognized object of %s",
+        fragmentManager.getClass());
+
+    if (sNoopFragmentManagerAccessor == null) {
+      sNoopFragmentManagerAccessor = new NoopFragmentManagerAccessor();
+    }
+
+    return sNoopFragmentManagerAccessor;
+  }
+
+  private static final class NoopFragmentAccessor implements FragmentAccessor {
+    @Override
+    public Object getFragmentManager(Object fragment) {
+      return null;
+    }
+
+    @Override
+    public int getId(Object fragment) {
+      return FragmentAccessor.NO_ID;
+    }
+
+    @Override
+    public String getTag(Object fragment) {
+      return null;
+    }
+
+    @Override
+    public View getView(Object fragment) {
+      return null;
+    }
+
+    @Override
+    public Object peekChildFragmentManager(Object fragment) {
+      return null;
+    }
+  }
+
+  private static final class NoopFragmentActivityAccessor implements FragmentActivityAccessor {
+    @Override
+    public Object getFragmentManager(Activity fragmentActivity) {
+      return null;
+    }
+
+    @Override
+    public Object getSupportFragmentManager(Activity fragmentActivity) {
+      return null;
+    }
+  }
+
+  private static final class NoopFragmentManagerAccessor implements FragmentManagerAccessor {
+    @Override
+    public List<?> getAddedFragments(Object fragmentManager) {
+      return null;
+    }
+  }
+
+  private static final class ReflectionFragmentAccessor implements FragmentAccessor {
+    // We access the field instead of calling the getChildFragmentManager() method
+    // because the latter will instantiate one if it doesn't exist
+    private final Field mFieldMChildFragmentManager;
+    private final Method mMethodGetFragmentManager;
+    private final Method mMethodGetId;
+    private final Method mMethodGetTag;
+    private final Method mMethodGetView;
+
+    public ReflectionFragmentAccessor(Class<?> fragmentClass) {
+      Util.throwIfNull(fragmentClass);
+
+      Field fieldMChildFragmentManager;
+      try {
+        fieldMChildFragmentManager = fragmentClass.getDeclaredField("mChildFragmentManager");
+        fieldMChildFragmentManager.setAccessible(true);
+      } catch (Exception e) {
+        LogUtil.w(
+            e,
+            "Could not retrieve mChildFragmentManager field from %s",
+            fragmentClass);
+
+        fieldMChildFragmentManager = null;
+      }
+
+      Method methodGetFragmentManager;
+      try {
+        methodGetFragmentManager = fragmentClass.getMethod("getFragmentManager");
+      } catch (Exception e) {
+        LogUtil.w(
+            e,
+            "Could not retrieve getFragmentManager() method from %s",
+            fragmentClass);
+
+        methodGetFragmentManager = null;
+      }
+
+      Method methodGetId;
+      try {
+        methodGetId = fragmentClass.getMethod("getId");
+      } catch (Exception e) {
+        LogUtil.w(
```

Comment:
Per our conversation, strongly prefer converting these to multicatches that catch all checked exceptions and rethrow using `throw new RuntimeException(ex);`.  But be careful, this advice only applies to methods that you expect exist (publicly documented, stable APIs).